File: app/src/main/java/com/veganbeauty/app/features/myskin/SkinHistoryAdapter.java
package com.veganbeauty.app.features.myskin;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.veganbeauty.app.databinding.ItemSkinHistoryBinding;

import org.json.JSONArray;
import org.json.JSONObject;

public class SkinHistoryAdapter extends RecyclerView.Adapter<SkinHistoryAdapter.ViewHolder> {

    public interface OnItemClickListener {
        void onItemClick(JSONObject item);
    }

    private JSONArray data;
    private final OnItemClickListener onItemClick;

    public SkinHistoryAdapter(JSONArray data, OnItemClickListener onItemClick) {
        this.data = data;
        this.onItemClick = onItemClick;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        ItemSkinHistoryBinding binding = ItemSkinHistoryBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false
        );
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        try {
            JSONObject item = data.getJSONObject(position);
            holder.bind(item);

            if (position == data.length() - 1) {
                holder.binding.itemSkinHistoryDivider.setVisibility(View.GONE);
            } else {
                holder.binding.itemSkinHistoryDivider.setVisibility(View.VISIBLE);
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    public int getItemCount() {
        return data != null ? data.length() : 0;
    }

    public void updateData(JSONArray newData) {
        this.data = newData;
        notifyDataSetChanged();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        public final ItemSkinHistoryBinding binding;

        public ViewHolder(ItemSkinHistoryBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(JSONObject item) {
            binding.itemSkinHistoryDate.setText(item.optString("date", ""));
            binding.itemSkinHistoryTime.setText(item.optString("time", ""));
            binding.itemSkinHistoryType.setText(item.optString("scanType", "Quét AI"));
            int score = item.optInt("score", -1);
            if (score < 0) {
                score = com.veganbeauty.app.data.local.SkinProfileMetricsHelper.computeOverallScore(item);
            }
            binding.itemSkinHistoryScore.setText(String.valueOf(score));

            binding.getRoot().setOnClickListener(v -> onItemClick.onItemClick(item));
        }
    }
}

File: app/src/main/java/com/veganbeauty/app/features/myskin/SkinHistoryAdapter.kt
package com.veganbeauty.app.features.myskin

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.veganbeauty.app.databinding.ItemSkinHistoryBinding
import org.json.JSONArray
import org.json.JSONObject

class SkinHistoryAdapter(
    private var data: JSONArray,
    private val onItemClick: (JSONObject) -> Unit
) : RecyclerView.Adapter<SkinHistoryAdapter.ViewHolder>() {

    inner class ViewHolder(val binding: ItemSkinHistoryBinding) : RecyclerView.ViewHolder(binding.root) {
        fun bind(item: JSONObject) {
            binding.itemSkinHistoryDate.text = item.optString("date", "")
            binding.itemSkinHistoryTime.text = item.optString("time", "")
            binding.itemSkinHistoryType.text = item.optString("scanType", "Quét AI")
            binding.itemSkinHistoryScore.text = item.optInt("score", 0).toString()

            binding.root.setOnClickListener {
                onItemClick(item)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val binding = ItemSkinHistoryBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return ViewHolder(binding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val item = data.getJSONObject(position)
        holder.bind(item)
        
        // Hide divider for the last item
        if (position == data.length() - 1) {
            holder.binding.itemSkinHistoryDivider.visibility = android.view.View.GONE
        } else {
            holder.binding.itemSkinHistoryDivider.visibility = android.view.View.VISIBLE
        }
    }

    override fun getItemCount(): Int {
        return data.length()
    }

    fun updateData(newData: JSONArray) {
        data = newData
        notifyDataSetChanged()
    }
}
